framework/src/eventloop/eventloop_loop.c

```diff
@@ -83,25 +83,24 @@ int eventloop_loop_run(void)
 		/* there are still events to process */
 		return EVENTLOOP_NOT_FINISHED;
 	}
-	release_app_loop();
 
-	return OK;
+	ret = release_app_loop();
+	if (ret != OK) {
+		eldbg("Running loop is done, but failed to close loop\n");
+	}
+
+	return ret;
 }
 
 int eventloop_loop_stop(void)
 {
-	el_loop_t *loop;
+	int index;
 
-	loop = get_app_loop();
-	if (loop == NULL) {
+	index = PID_HASH(getpid());
```

Comment:
How about use another MACRO instead of ```PID_HASH(getpid())```?
like get_this_thread_loop_idx?

Reply:
Yes, I agree with your opinion. I will update all later.